samples/bluetooth/direction_finding_peripheral/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <zephyr.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/bas.h>
+#include <bluetooth/services/hrs.h>
+#include <bluetooth/direction.h>
+
+#define DF_FEAT_ENABLED BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP)
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_LE_SUPPORTED_FEATURES, BT_LE_SUPP_FEAT_24_ENCODE(DF_FEAT_ENABLED)),
+};
+
+/* Latency set to zero, to enforce PDU exchange every connection event */
+#define CONN_LATENCY 5U
+/* Inteval used to run CTE request procedure periodically.
+ * Value is a number of connection events.
+ */
+#define CTE_REQ_INTERVAL (CONN_LATENCY + 10U)
+/* Length of CTE in unit of 8 us */
+#define CTE_LEN (0x14U)
+
+#if defined(CONFIG_BT_CTLR_DF_ANT_SWITCH_RX)
+static const uint8_t ant_patterns[] = { 0x2, 0x0, 0x5, 0x6, 0x1, 0x4, 0xC, 0x9, 0xE, 0xD, 0x8 };
```

Comment:
is there an explanation somewhere of what the values mean?

Reply:
The numbers are arbitrary. They have a meaning if there is a particular antenna matrix used with sample. So numbers are vendor specific. Here are just some numbers that are used by Controller to change state of GPIOs.